File: frontend/src/types/bot.ts
/**
 * Типы для ботов
 */

export type BotStatus = 'active' | 'inactive' | 'suspended';

export interface BotResponse {
    id: string;
    name: string;
    username: string;
    description: string | null;
    about: string | null;
    owner_id: string;
    status: BotStatus;
    api_token: string | null;
    created_at: string;
}

export interface BotCommand {
    command: string;
    description: string;
}

export interface BotButton {
    text: string;
    callback_data?: string;
    url?: string;
}

export interface BotKeyboard {
    buttons: BotButton[][];
    resize?: boolean;
    one_time?: boolean;
}

export interface InlineButton {
    text: string;
    callback_data?: string;
    url?: string;
    switch_inline?: string;
}

export interface InlineKeyboard {
    buttons: InlineButton[][];
}

export interface CreateBotRequest {
    name: string;
    username: string;
    description?: string;
    about?: string;
}

export interface BotMessageRequest {
    chat_id: string;
    text: string;
    keyboard?: BotKeyboard;
    inline_keyboard?: InlineKeyboard;
    reply_to_message_id?: string;
}

File: backend/src/schemas/bot.py
"""Схемы для ботов."""

import uuid
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class BotStatus(str, Enum):
    """Статус бота."""

    ACTIVE = "active"
    INACTIVE = "inactive"
    SUSPENDED = "suspended"


class CommandType(str, Enum):
    """Тип команды бота."""

    TEXT = "text"
    BUTTON = "button"
    INLINE = "inline"


class CreateBotRequest(BaseModel):
    """Запрос на создание бота."""

    name: str = Field(..., min_length=1, max_length=64)
    username: str = Field(..., min_length=3, max_length=32, pattern=r"^[a-zA-Z0-9_]+$")
    description: str | None = Field(None, max_length=500)
    about: str | None = Field(None, max_length=200)


class BotResponse(BaseModel):
    """Ответ с информацией о боте."""

    id: uuid.UUID
    name: str
    username: str
    description: str | None
    about: str | None
    owner_id: uuid.UUID
    status: BotStatus
    api_token: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class BotCommand(BaseModel):
    """Команда бота."""

    command: str = Field(..., pattern=r"^[a-z0-9_]+$", max_length=32)
    description: str = Field(..., max_length=256)


class BotCommandsRequest(BaseModel):
    """Запрос на установку команд бота."""

    commands: list[BotCommand]


class BotButton(BaseModel):
    """Кнопка бота."""

    text: str = Field(..., max_length=64)
    callback_data: str | None = Field(None, max_length=64)
    url: str | None = None


class BotKeyboard(BaseModel):
    """Клавиатура бота."""

    buttons: list[list[BotButton]]
    resize: bool = True
    one_time: bool = False


class InlineButton(BaseModel):
    """Инлайн-кнопка."""

    text: str = Field(..., max_length=64)
    callback_data: str | None = None
    url: str | None = None
    switch_inline: str | None = None


class InlineKeyboard(BaseModel):
    """Инлайн-клавиатура."""

    buttons: list[list[InlineButton]]


class BotMessageRequest(BaseModel):
    """Запрос на отправку сообщения от бота."""

    chat_id: uuid.UUID
    text: str = Field(..., max_length=4000)
    keyboard: BotKeyboard | None = None
    inline_keyboard: InlineKeyboard | None = None
    reply_to_message_id: uuid.UUID | None = None


class BotMessageResponse(BaseModel):
    """Ответ после отправки сообщения."""

    message_id: uuid.UUID
    chat_id: uuid.UUID
    sent_at: datetime


class WebhookConfig(BaseModel):
    """Конфигурация вебхука."""

    url: str
    secret: str | None = None
    allowed_updates: list[str] | None = None


class BotUpdateRequest(BaseModel):
    """Запрос на обновление бота."""

    name: str | None = None
    description: str | None = None
    about: str | None = None
